ovirt_imageio/client/_api.py

```diff
@@ -69,6 +69,10 @@ def upload(filename, url, cafile, buffer_size=BUFFER_SIZE, secure=True,
             image data, leaving unallocated areas as holes, exposing data from
             the target disk backing chain. Valid only when uploding to an empty
             snapshot.
+        disk_is_zero (bool): If set, skip zero extents in the source image
+            instead of zeroing the extent in the destination disk. Should be
+            used only when uploading to new disk on file based storage or new
+            sparse disk on block based storage.
```

Comment:
**At first glance** is a bit weird to read the parameter name, and then do `zero=not (disk_is_zero [...]` below. I do understand the logic: if the full disk is zero, then we can skip zero extents, so we do not need to zero the area on destination. But maybe another name would avoid confusion? What about `disk_is_empty`?

Also, I am curious about what would happen if we actually skip zero extents when uploading to non-empty disks? Will the disk be corrupted? Will it work but the disks would not be the same when compared?

Reply:
disk_is_empty is not correct for raw preallocated disk on block storage - the disk is empty, but when we read from it we may get data from previous use of the device. It is more correct to use disk_is_zero, meaning that reading from the disk will return zeros.

In qemu-img convert this option is called --target-is-zero. The client disk_is_zero try to be similar to qemu-img.

If you don't zero a raw disk you may have random data on the disk. If you create a file system on the disk the data is likley to be hidden, but if you use the disk directly (e.g. database using raw block device) you will have junk data that may break the application.

We also have similar flag in vdsm copy_data.

Reply:
The docstring needs a fix, it should be:

    should be used only when uploading to new disk on file based storage or
    new qcow2 disk on block based storage.

Reply:
Ah! I get it. Thanks for the explanations.

And the justification for keeping the name is good too. Even if only for similarity with `qemu-img`, already good reason.